File: simple/csharp/extract-method_isolate-switch_before.cs
class Order {
// ...

double getTotal() {
  double total = 0;
  for (Product product : getProducts()) {
    total = product.quantity * product.price;
  }

  // Apply regional discounts.
  switch (user.getCountry()) {
    'US': total *= 0.85; break;
    'RU': total *= 0.75; break;
    'CN': total *= 0.9; break;
    // ...
  }

  return total;
}

File: simple/php/extract-method_isolate-switch_before.php
class Order {
// ...

function calculateTotal() {
  $total = 0;
  foreach ($this->getProducts() as $product) {
    $total = $product->quantity * $product->price;
  }

  // Apply regional discounts.
  switch ($this->user->getCountry()) {
    'US': $total *= 0.85; break;
    'RU': $total *= 0.75; break;
    'CN': $total *= 0.9; break;
    // ...
  }

  return $total;
}